src/main/kotlin/io/github/mojira/arisa/ArisaMain.kt

```diff
@@ -1,165 +1,68 @@
 package io.github.mojira.arisa
 
-import arrow.syntax.function.partially1
 import ch.qos.logback.classic.AsyncAppender
 import ch.qos.logback.classic.LoggerContext
 import com.github.napstr.logback.DiscordAppender
 import com.uchuhimo.konf.Config
 import com.uchuhimo.konf.source.yaml
-import io.github.mojira.arisa.domain.Issue
-import io.github.mojira.arisa.infrastructure.HelperMessageService
-import io.github.mojira.arisa.infrastructure.ProjectCache
 import io.github.mojira.arisa.infrastructure.config.Arisa
-import io.github.mojira.arisa.infrastructure.jira.connectToJira
-import io.github.mojira.arisa.infrastructure.jira.toDomain
-import io.github.mojira.arisa.registry.TicketQueryTimeframe
-import io.github.mojira.arisa.registry.getModuleRegistries
 import net.rcarz.jiraclient.JiraClient
 import org.slf4j.Logger
 import org.slf4j.LoggerFactory
-import java.io.File
-import java.lang.Long.max
-import java.time.Duration
-import java.time.Instant
-import java.time.temporal.ChronoUnit
 import java.util.concurrent.TimeUnit
 
 val log: Logger = LoggerFactory.getLogger("Arisa")
 
-private const val TIME_MINUTES = 5L
-private const val MAX_QUERY_TIMEFRAME_IN_MINUTES = 10L
-
-const val MAX_RESULTS = 50
+lateinit var jiraClient: JiraClient
 
-private const val RELOG_INTERVAL_IN_SECONDS = 5 * 60L
-private const val WAIT_TIME_AFTER_CONNECTION_ERROR_IN_SECONDS = 40L
+private val config = readConfig()
+private val connectionService = JiraConnectionService(config)
+private var executor = Executor(config)
 
-lateinit var jiraClient: JiraClient
+const val MIN_TIME_BETWEEN_EXECUTION_CYCLES_IN_SECONDS = 10L
```

Comment:
We can use 10.seconds here (yay for kotlin)

I think its out of experimental

Reply:
Yeah, that works. Might be nice to change it in the future, but that'd be another larger change since that'd cascade down to other things too I think.

Reply:
Looks like it is still experimental, see https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.time/
Though it would indeed be good to refactor this and other time related checks in the future. Personally I find these `plus(x).isBefore(...)` checks more difficult to understand than using JDK `Duration`, e.g. `Duration.between(...).toMinutes() < ...`.

Reply:
Hm, personally I have troubles with `Duration.between` because it might be negative and do unexpected things... so just to be sure I always put `.abs()` there which is annoying and I don't like it. Not sure if there's a better option.

Reply:
Ah, also it seems like what I experimented with was arrow's version of the `.seconds` suffix, so that was probably something different than you meant.